test/clientSpec.js

```diff
@@ -99,62 +134,85 @@ describe('Asymmetric plugin virtual', () => {
     })
 
     it('should receive and fulfill a transfer', async function () {
-      this.transfer.to = this.plugin.getAccount()
+      this.transferJson.to = this.plugin.getAccount() // TODO: needed?
+
+      this.mockSocket
+        .reply(clpPacket.TYPE_RESPONSE, ({requestId, data}) => {
+          return clpPacket.serializeResponse(requestId, [])
+        })
+        .reply(clpPacket.TYPE_FULFILL, ({requestId, data}) => {
+          assert.equal(data.transferId, this.transferJson.id)
+          assert.equal(data.fulfillment, this.fulfillment)
+          return clpPacket.serializeResponse(requestId, [])
+        })
+
       const prepared = new Promise((resolve) =>
         this.plugin.once('incoming_prepare', () => resolve()))
 
-      await this.plugin.receive('send_transfer', [ this.transfer ])
+      await this.plugin.receive(this.mockSocket, this.transfer)
       await prepared
 
-      nock('https://example.com')
-        .post('/rpc?method=fulfill_condition&prefix=example.red.',
-          [ this.transfer.id, this.fulfillment ])
-        .reply(200, true)
-
       const fulfilled = new Promise((resolve) =>
         this.plugin.once('incoming_fulfill', () => resolve()))
 
-      await this.plugin.fulfillCondition(this.transfer.id, this.fulfillment)
+      await this.plugin.fulfillCondition(this.transferJson.id, this.fulfillment)
       await fulfilled
     })
 
     it('should not send a transfer if peer gives error', async function () {
-      nock('https://example.com')
-        .post('/rpc?method=send_transfer&prefix=example.red.', [ this.transfer ])
-        .reply(500)
+      this.mockSocket
+        .reply(clpPacket.TYPE_PREPARE, ({requestId, data}) => {
+          const expectedPacket = clpPacket.deserialize(this.transfer)
+          assert.deepEqual(data, expectedPacket.data)
+
+          const ilp = ilpPacket.serializeIlpError({
+            code: 'F00',
+            name: 'Bad Request',
+            triggeredBy: '',
+            forwardedBy: [],
+            triggeredAt: new Date(),
+            data: JSON.stringify({ message: 'Peer isn\'t feeling like it.' })
+          })
+
+          return clpPacket.serializeError({rejectionReason: ilp}, requestId, [])
+        })
 
       const prepared = new Promise((resolve, reject) => {
-        this.plugin.once('outgoing_prepare', () =>
-          reject(new Error('should not be accepted')))
+        this.plugin.once('outgoing_prepare', () => {
+          reject(new Error('should not be accepted'))
+        })
         setTimeout(resolve, 10)
       })
 
-      await assert.isRejected(this.plugin.sendTransfer(this.transfer))
+      await assert.isRejected(this.plugin.sendTransfer(this.transferJson))
       await prepared
     })
   })
 
   describe('server', function () {
-    it('should call several plugins over RPC', async function () {
+    // TODO: Is this test case still relevant?  (cc: sharafian)
+    // Would you ever add several several sockets to a single plugin?
```

Comment:
if there are several clients subscribed to one server (i.e. a user has lots of apps that are all authenticated with their provider), then the provider's server plugin will have several sockets